### DISCUSSION TITLE
How can I customize my company's devices in zha-device-handlers to send commands to private clusters

### DISCUSSION THREAD

**Discussion by hwzolin** [zigpy/zha-device-handlers#3060] [Q&A]
Hi all, it's my first time to add a custom device on ZHA. Our company's PLUG has a private cluster that is used to zero the value of summation delivered, the cluster id is 0xFF03 and the attribute id is 0x0000. 
How do I create a new class for sending this command?

**Comment by Hedda** [comment]
Don't think devs watch this section so you might need to post to zigpy repository to get some attention -> https://github.com/zigpy/zigpy

**Reply by dmulcahey** [reply]
please stop telling folks to cross post

**Comment by dmulcahey** [comment]
> Hi all, it's my first time to add a custom device on ZHA. Our company's PLUG has a private cluster that is used to zero the value of summation delivered, the cluster id is 0xFF03 and the attribute id is 0x0000. How do I create a new class for sending this command?

Define the custom cluster with its attributes like this: https://github.com/zigpy/zha-device-handlers/blob/4b3d6591c72302fd82e1b5a6e4ee30eb2e08dd9c/zhaquirks/schneiderelectric/dimmers.py#L43

Once that is defined in the cluster correctly we can add a V2 quirk that will expose a button that can be used to write the attribute like this: https://github.com/zigpy/zigpy/blob/3d1b60c9181fb40638eaf3e70c5815967b07300c/tests/test_quirks_v2.py#L545

**Reply by hwzolin** [reply]
Thank you very much for your patient reply. I will try to implement it according to the ideas you provided.
